### PR TITLE
To Do list: add & remove

### PR DESCRIPTION
**Pull request by GiuseppeTG**
In this feature I did the following:

- Removed all hardcoded items from the tasks array.

- Created a new JavaScript file for the new functionality.

- Implemented a function for adding a new task (add a new element to the array).

- Implemented a function for deleting a task (remove an element from the array).

- Implemented a function for editing task descriptions.

- Deleting a task updates all remaining items' indexes, so they represent the current list order and are unique.

- All changes to the To-Do List are saved in local storage.

### CODE REVIEW

File: src/modules/storage.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    let newArr = [];
+    for (let i = 0; i < tasks.length; i += 1) {
+      if (tasks[i].completed === taskToDelete) {
+        newArr = tasks.filter((task) => task.completed === false);
+      }
+    }
+
+    for (let i = 0; i < newArr.length; i += 1) {
+      newArr[i].index = i;
+    }
```
Comment (roshan-bajgain) [inline src/modules/storage.js:30]: - [ ] The property index is set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal to 5). So the index starts from 1, not from 0. Kindly implement it.

**Review by roshan-bajgain — CHANGES_REQUESTED** [review]
Hi @GiuseppeTG, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Professional README file. ✔️
- PR has a good title and summary. ✔️
- Your app is working properly. Good job ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by GiuseppeTG** [comment]
Hello @roshan-bajgain! 
I'm trying to implement that but when I do it all the functionalities stop working. I don't know if by any chance this project can be done with the indexation of each task to at 0. 
It is the only thing I am missing for this project, and this implementation does not affect the result or the UI at all. 
I'm trying very hard to make it work with index 0, and for making it work a should rewrite almost all my code 😞. It is almost the end of the deadline and it would be a shame to repeat a week just because of something like that. 
I would appreciate if you tell me if a can talk with someone for this problem.

**Review by RolandM99 — APPROVED** [review]
<h1>Status: Approve 🥇 ✔</h1>
Hi @GiuseppeTG, 

### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by roshan-bajgain** [comment]
> Hello @roshan-bajgain! I'm trying to implement that but when I do it all the functionalities stop working. I don't know if by any chance this project can be done with the indexation of each task to at 0. It is the only thing I am missing for this project, and this implementation does not affect the result or the UI at all. I'm trying very hard to make it work with index 0, and for making it work a should rewrite almost all my code 😞. It is almost the end of the deadline and it would be a shame to repeat a week just because of something like that. I would appreciate if you tell me if a can talk with someone for this problem.

@GiuseppeTG I am sorry that I cannot reply on time, I saw your project just now and it was not done so you can implement it.
In your test.js file, in line number 5 just increase the index by 1 eg: (index + 1) and then it will work. Thank you
### solution
ll

**Pull request merged**